builtin/logical/pki/path_roles.go

```diff
@@ -550,57 +545,59 @@ func parseKeyUsages(input []string) int {
-	EnforceHostnames              bool     `json:"enforce_hostnames" mapstructure:"enforce_hostnames"`
-	AllowIPSANs                   bool     `json:"allow_ip_sans" mapstructure:"allow_ip_sans"`
-	ServerFlag                    bool     `json:"server_flag" mapstructure:"server_flag"`
-	ClientFlag                    bool     `json:"client_flag" mapstructure:"client_flag"`
-	CodeSigningFlag               bool     `json:"code_signing_flag" mapstructure:"code_signing_flag"`
-	EmailProtectionFlag           bool     `json:"email_protection_flag" mapstructure:"email_protection_flag"`
-	UseCSRCommonName              bool     `json:"use_csr_common_name" mapstructure:"use_csr_common_name"`
-	UseCSRSANs                    bool     `json:"use_csr_sans" mapstructure:"use_csr_sans"`
-	KeyType                       string   `json:"key_type" mapstructure:"key_type"`
-	KeyBits                       int      `json:"key_bits" mapstructure:"key_bits"`
-	MaxPathLength                 *int     `json:",omitempty" mapstructure:"max_path_length"`
-	KeyUsageOld                   string   `json:"key_usage,omitempty"`
-	KeyUsage                      []string `json:"key_usage_list" mapstructure:"key_usage"`
-	OUOld                         string   `json:"ou,omitempty"`
-	OU                            []string `json:"ou_list" mapstructure:"ou"`
-	OrganizationOld               string   `json:"organization,omitempty"`
-	Organization                  []string `json:"organization_list" mapstructure:"organization"`
-	Country                       []string `json:"country" mapstructure:"country"`
-	Locality                      []string `json:"locality" mapstructure:"locality"`
-	Province                      []string `json:"province" mapstructure:"province"`
-	StreetAddress                 []string `json:"street_address" mapstructure:"street_address"`
-	PostalCode                    []string `json:"postal_code" mapstructure:"postal_code"`
-	GenerateLease                 *bool    `json:"generate_lease,omitempty"`
-	NoStore                       bool     `json:"no_store" mapstructure:"no_store"`
-	RequireCN                     bool     `json:"require_cn" mapstructure:"require_cn"`
-	AllowedOtherSANs              []string `json:"allowed_other_sans" mapstructure:"allowed_other_sans"`
-	PolicyIdentifiers             []string `json:"policy_identifiers" mapstructure:"policy_identifiers"`
-	BasicConstraintsValidForNonCA bool     `json:"basic_constraints_valid_for_non_ca" mapstructure:"basic_constraints_valid_for_non_ca"`
+	LeaseMax                      string        `json:"lease_max"`
+	Lease                         string        `json:"lease"`
+	MaxTTL                        string        `json:"max_ttl" mapstructure:"max_ttl"`
+	TTL                           string        `json:"ttl" mapstructure:"ttl"`
+	TTLDuration                   time.Duration `json:"ttl_duration" mapstructure:"ttl_duration"`
```

Comment:
Many of the other "upgraded" fields rename the struct field to <fieldname>Old but keeps the json tags the same.  See AllowedDomains as an example.  There is nothing wrong with this approach but it may be easier to understand if they all used the same convention.

Reply:
Up to you really, just pointing it out.  I'm fine either way.